modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRow.es.js

```diff
@@ -0,0 +1,326 @@
+import React, {Component} from 'react';
+import {PropTypes} from 'prop-types';
+import ClayButton from '../shared/ClayButton.es';
+import ClayIcon from '../shared/ClayIcon.es';
+import ClaySelect from '../shared/ClaySelect.es';
+import {CONJUNCTIONS} from '../../utils/constants.es';
+import {DragSource as dragSource, DropTarget as dropTarget} from 'react-dnd';
+import {DragTypes} from '../../utils/drag-types.es';
+import getCN from 'classnames';
+import {generateGroupId, sub} from '../../utils/utils.es';
+
+const acceptedDragTypes = [
+	DragTypes.CRITERIA_ROW,
+	DragTypes.PROPERTY
+];
+
+/**
+ * Prevents rows from dropping onto itself.
+ * This method must be called `canDrop`.
+ * @param {Object} props Component's current props.
+ * @param {DropTargetMonitor} monitor
+ * @returns {boolean} True if the target should accept the item.
+ */
+function canDrop(props, monitor) {
```

Comment:
👍 great util

Reply:
This is actually part of `react-dnd` 😅 
http://react-dnd.github.io/react-dnd/docs/api/drop-target